packages/backend-dev-utils/src/DevDataStore.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BackstageIpcClient, ipcClient } from './ipcClient';
+
+interface SaveRequest {
+  key: string;
+  data: unknown;
+}
+
+interface SaveResponse {
+  saved: boolean;
+}
+
+interface LoadRequest {
+  key: string;
+}
+
+interface LoadResponse {
+  loaded: boolean;
+  data: unknown;
+}
+
+/**
+ * A data store that can be used to store temporary data during development.
+ *
+ * @public
+ */
+export class DevDataStore {
+  static #instance?: DevDataStore;
+
+  /**
+   * Tries to acquire a DevDataStore instance. This will only succeed when the backend
+   * process is being run through the `@backstage/cli` in development mode.
+   *
+   * @returns A DevDataStore instance, or undefined if not available.
+   */
+  static get(): DevDataStore | undefined {
+    if (ipcClient) {
+      if (!this.#instance) {
+        this.#instance = new DevDataStore(ipcClient);
+      }
+      return this.#instance;
+    }
+    return undefined;
+  }
+
+  /** @internal */
+  static forTest(client: Pick<BackstageIpcClient, 'request'>): DevDataStore {
+    return new DevDataStore(client as BackstageIpcClient);
```

Comment:
Will it ever be useful to set this.#instance with this one as well? So that "deeper" code that is a transitive bit of code to the one being tested can use `get()` and successfully get this one. Maybe that doesn't quite apply here 🤔